### PR TITLE
Restrict K8s role definitions to actually needed roles

### PR DESCRIPTION
**Pull request by piotrkmita**
Fixes [#255](https://github.com/druid-io/druid-operator/issues/225).

- daemonsets, replicasets and servicemonitors, finalizers removed - usage of them are missing
- extensions/ingress removed - it has been removed for a while (https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/)
- rest of resources changed from `*` to an explicit list.
<hr>

This PR has:
- [X] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [X] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `role.yaml`

### CODE REVIEW

File: deploy/role.yaml
Comment (AdheipSingh) [inline deploy/role.yaml:84]: nit: format

**Comment by AdheipSingh** [comment]
@piotrkmita thanks for the PR. 

> daemonsets, replicasets and servicemonitors, finalizers removed - usage of them are missing

druid operator has a finalizer implementation, did you test the PR with the default behaviour of the finalizer ( enabled by default ) ? 
BTW what version of k8s was this PR testing on ?

Thanks

**Comment by piotrkmita** [comment]
Hey @AdheipSingh,

Our k8s team is not allowing setting finalizers on built-in types. This is the response we've got from them.
> The finalizers should only be on the custom resource. I don't think you can even set finalizers on built-in types, so it does not make sense to allow it in RBAC, which is why we don't allow this.

That's why we've tried default behavior which worked fine and in the operator logs we could see messages from finalizer
image
We've also tried using `disablePVCDeletionFinalizer` parameter to make sure it's working properly.

We’ve tested it with both minikube and our test cluster. Here are the versions:
minikube:
`Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64”}`

test cluster:
`Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:07Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}`

Let me know if we missed something there,
Thanks!

File: deploy/role.yaml
Comment (piotrkmita) [reply deploy/role.yaml:84]: Just to make sure, is it about newline missing?

Comment (AdheipSingh) [reply deploy/role.yaml:84]: yes :)

**Comment by AdheipSingh** [comment]
@piotrkmita 
thanks for the info. 
LGTM , though ill run a test on in house cluster and revert back.

**Review by AdheipSingh — APPROVED** [review]

**Review by nishantmonu51 — APPROVED** [review]
👍

File: chart/templates/role.yaml
Comment (AdheipSingh) [inline chart/templates/role.yaml:34]: add ```patch``` too for events. 
this will be needed for this PR  https://github.com/druid-io/druid-operator/pull/223
currently operator's creates events each time, with this https://github.com/druid-io/druid-operator/pull/223 a re-occurring event will be patched.

**Review by AdheipSingh — CHANGES_REQUESTED** [review]

**Comment by AdheipSingh** [comment]
have tested this PR on v1.21.2 do and 1.19 gke.
LGTM
Thanks @piotrkmita 

@himanshug @nishantmonu51 good to go :)

**Review by AdheipSingh — APPROVED** [review]

**Review by nishantmonu51 — APPROVED** [review]
👍

**Comment by AdheipSingh** [comment]
@himanshug waiting for your approval :)

**Comment by himanshug** [comment]
thanks, took a cursory look, LGTM

**Pull request merged**
